web/frontend/src/components/SearchBox.vue

```diff
@@ -123,11 +125,22 @@ export default defineComponent({
       } else {
         url = `/pelias/v1/autocomplete?text=${encodeURIComponent(value)}`;
       }
+      const thisRequestIdx = requestIdx;
+      requestIdx++;
       const response = await fetch(url);
       if (response.status != 200) {
-        autocompleteOptions.value = [];
+        if (thisRequestIdx > mostRecentResultsRequestIdx) {
+          autocompleteOptions.value = [];
+        }
+        return;
+      }
+      if (thisRequestIdx < mostRecentResultsRequestIdx) {
+        // console.debug(`not updating autocomplete for req #${thisRequestIdx} because req #${mostRecentResultsRequestIdx} completed in the meanwhile.`);
```

Comment:
I can delete these commented out log lines if you'd prefer, but they also serve as decent documentation. Either way!

Reply:
I generally don't like commented out code. Old habit from my google days to avoid doing that when I notice it. Definitely a nit though, this looks good overall. Documentation is good though so feel free to leave the message as a comment, just no `console.debug`. :)

Reply:
done!